vsphere/datadog_checks/vsphere/vsphere.py

```diff
@@ -210,16 +214,17 @@ def submit_metrics_callback(self, query_results):
                         to_string(metric_name),
                     )
                     continue
-                value = valid_values[-1]
-                if metric_name in PERCENT_METRICS:
-                    # Convert the percentage to a float.
-                    value /= 100.0
 
                 tags = []
-                if should_collect_per_instance_values(metric_name, resource_type):
-                    instance_tag_key = get_mapped_instance_tag(metric_name)
-                    instance_tag_value = result.id.instance or 'none'
-                    tags.append('{}:{}'.format(instance_tag_key, instance_tag_value))
+                if should_collect_per_instance_values(self.config, metric_name, resource_type):
+                    instance_value = result.id.instance
+                    # When collecting per instance values, it's possible that both aggregated metric and per instance
+                    # metrics are received. In that case, the metric with no instance value is skipped.
+                    if is_metric_available_per_instance(metric_name, resource_type) and not instance_value:
```

Comment:
```suggestion
                    if not instance_value:
```
You already checked that the metric is available per-instance with `should_collect_per_instance_values`

Reply:
Actually you can just remove that whole if block

Reply:
Good catch. It was previously needed but not anymore after some refactoring.